pkg/deploy/generator/templates_gateway.go

```diff
@@ -0,0 +1,147 @@
+package generator
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"github.com/Azure/ARO-RP/pkg/env"
+	"github.com/Azure/ARO-RP/pkg/util/arm"
+)
+
+func (g *generator) gatewayManagedIdentityTemplate() *arm.Template {
+	t := templateStanza()
+
+	t.Resources = append(t.Resources,
+		g.gatewayManagedIdentity(),
+	)
+
+	return t
+}
+
+func (g *generator) gatewayTemplate() *arm.Template {
+	t := templateStanza()
+
+	params := []string{
+		"acrResourceId",
+		"databaseAccountName",
+		"dbTokenUrl",
+		"encryptionAtHost",
+		"gatewayDomains",
+		"gatewayFeatures",
+		"gatewayMdsdConfigVersion",
+		"gatewayServicePrincipalId",
+		"gatewayStorageAccountDomain",
+		"gatewayVmssCapacity",
+		"keyvaultPrefix",
+		"mdmFrontendUrl",
+		"mdsdEnvironment",
+		"rpImage",
+		"rpResourceGroupName",
+		"rpServicePrincipalId",
+		"sshPublicKey",
+		"vmSize",
+		"vmssName",
+	}
+
+	for _, param := range params {
+		p := &arm.TemplateParameter{Type: "string"}
+		switch param {
+		case "encryptionAtHost":
+			p.Type = "bool"
+		case "gatewayDomains",
+			"gatewayFeatures":
+			p.DefaultValue = ""
+		case "gatewayVmssCapacity":
+			p.Type = "int"
+			p.DefaultValue = 3
+		case "vmSize":
+			p.DefaultValue = "Standard_D2s_v3"
+		}
+		t.Parameters[param] = p
+	}
+
+	t.Resources = append(t.Resources,
+		g.gatewayStorageAccount(),
+		g.gatewayLB(),
+		g.gatewayPLS(),
+		g.gatewayVMSS(),
```

Comment:
add TODO to use rpLBAlert() to define Azure monitoring alerts around the readiness of the gateway ILB